rules.d/90-image-dissect.rules

```diff
@@ -0,0 +1,25 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+ACTION=="remove", GOTO="image_dissect_end"
+SUBSYSTEM!="block", GOTO="image_dissect_end"
+
+# add symlink to GPT root disk
+ENV{ID_PART_GPT_AUTO_ROOT}=="1", ENV{ID_FS_TYPE}!="crypto_LUKS", SYMLINK+="gpt-auto-root"
+ENV{ID_PART_GPT_AUTO_ROOT}=="1", ENV{ID_FS_TYPE}=="crypto_LUKS", SYMLINK+="gpt-auto-root-luks"
+ENV{DM_UUID}=="CRYPT-*", ENV{DM_NAME}=="root", SYMLINK+="gpt-auto-root"
+
+# If this is the whole disk that we booted from, then dissect it
+ENV{DEVTYPE}=="disk", ENV{ID_PART_GPT_AUTO_ROOT_ORIGIN}=="1", IMPORT{program}="dissect_id $devnode"
+# If this is a partition, and we found something on the parent, then copy the right properties from the parent, and rename them
+ENV{DEVTYPE}=="partition", ENV{ID_DISSECT_IMAGE}!="", IMPORT{program}="dissect_id --copy $devnode"
+
+ENV{DEVTYPE}=="partition", ENV{ID_DISSECT_PART_DESIGNATOR}!="", SYMLINK+="gpt-auto-$env{ID_DISSECT_PART_DESIGNATOR}"
```

Comment:
I still think disk/by-parttype/ as implemented in #28894 is way more descriptive and well-organized than laying all these directly in /dev/

Reply:
well, my goal here was to not introduce new naming, but to just mildly extend the naming of the existing /dev/gpt-auto-root symlink

Reply:
or to say this differently, this is a natural extension of the existing /dev/gpt-auto-root  logic, just with a bunch more partitions.

i think the /dev/disk/by-* is kinda important if we basically unbounded number of devices in there. But this is different, the number of symlinks we'll create here is bounded by the number of designators, which is pretty low. hence I think we can get away with following the current naming scheme and extending it a tiny bit for the other designators, and don't have to introduce something entirely new.